api/openapi/lookups_api.py

```diff
@@ -0,0 +1,94 @@
+import json
+
+from django.conf import settings
+from django.http import JsonResponse
+from rest_framework import viewsets
+from rest_framework.exceptions import (
+    APIException,
+    MethodNotAllowed,
+    NotFound,
+    ValidationError,
+)
+
+
+class UniversityLookupAPI(viewsets.GenericViewSet):
+    """
+    The university lookup API with read-only methods to get university information.
+    """
+
+    def __get_universities(self):
+        """Get the universities data from the json file."""
+        with open(settings.STATICFILES_DIRS[0] + "/common/universities.json", "r") as f:
+            universities = json.load(f).get("universities", dict())
+            list = [{"code": k, "name": v} for k, v in universities.items()]
+            return list
+
+    def list_or_single(self, request):
+        """
+        This method handles the /universities endpoint.
+        With a query string get parameter "code" it returns a single university.
+        Without a query string parameter it returns the entire list.
+
+        :returns list of dict or dict: The dict contains attributes code and name.
+        """
+        print("UniversityLookupAPI. Entered list_or_single")
+
+        try:
+            universities = self.__get_universities()
+        except Exception as e:
+            print(f"Exception: {e}")
```

Comment:
I wish we would at some point be doing proper logging.

But with current setup it's good enough :)

Reply:
I completely agree about adding real logging. And auditing.